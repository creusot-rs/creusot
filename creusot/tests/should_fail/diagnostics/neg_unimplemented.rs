#![allow(unused)]
extern crate creusot_contracts;
use creusot_contracts::*;

#[logic]
fn negation(x: i32) -> i32 {
    // FIXME(diagnostics): Why is the error reported twice?
    -x
}
