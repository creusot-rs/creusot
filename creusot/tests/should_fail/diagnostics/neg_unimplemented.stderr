error[E0277]: cannot apply unary operator `-` to type `i32`
 --> neg_unimplemented.rs:8:6
  |
8 |     -x
  |     -^
  |     ||
  |     |cannot apply unary operator `-` in logic
  |     required by a bound introduced by this call
  |
  = help: the trait `creusot_contracts::logic::ops::NegLogic` is not implemented for `i32`
  = help: the trait `creusot_contracts::logic::ops::NegLogic` is implemented for `creusot_contracts::Int`

error[E0277]: cannot apply unary operator `-` to type `i32`
 --> neg_unimplemented.rs:8:5
  |
8 |     -x
  |     ^^ cannot apply unary operator `-` in logic
  |
  = help: the trait `creusot_contracts::logic::ops::NegLogic` is not implemented for `i32`
  = help: the trait `creusot_contracts::logic::ops::NegLogic` is implemented for `creusot_contracts::Int`

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0277`.
